From 9b58077ff3fe6025cf22ec002541a7513a5a05f9 Mon Sep 17 00:00:00 2001
From: Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
Date: Sun, 22 Feb 2015 19:51:13 +0000
Subject: [PATCH] Fix build on general purpose distros

General purpose distros do not build with Raspberry Pi mkspecs, hence
we need to manually add include directories and libraries.
---
 config.tests/brcm_egl/brcm_egl.pro                               | 4 ++--
 src/hardwareintegration/client/brcm-egl/brcm-egl.pri             | 4 ++--
 src/hardwareintegration/compositor/brcm-egl/brcm-egl.pri         | 4 ++--
 src/plugins/hardwareintegration/client/brcm-egl/brcm-egl.pro     | 2 --
 src/plugins/hardwareintegration/compositor/brcm-egl/brcm-egl.pro | 2 --
 src/plugins/platforms/qwayland-brcm-egl/qwayland-brcm-egl.pro    | 2 --
 6 files changed, 6 insertions(+), 12 deletions(-)

diff --git a/config.tests/brcm_egl/brcm_egl.pro b/config.tests/brcm_egl/brcm_egl.pro
index a6b416f..c1fdadc 100644
--- a/config.tests/brcm_egl/brcm_egl.pro
+++ b/config.tests/brcm_egl/brcm_egl.pro
@@ -1,13 +1,13 @@
 TARGET = brcm_egl
 CONFIG -= qt
 
-INCLUDEPATH += $$QMAKE_INCDIR_EGL
+INCLUDEPATH += /opt/vc/include /opt/vc/include/interface/vcos/pthreads
 
 for(p, QMAKE_LIBDIR_EGL) {
     exists($$p):LIBS += -L$$p
 }
 
-LIBS += $$QMAKE_LIBS_EGL
+LIBS += -L/opt/vc/lib -lGLESv2 -lEGL -lm -lbcm_host
 
 # Input
 SOURCES += main.cpp
diff --git a/src/hardwareintegration/client/brcm-egl/brcm-egl.pri b/src/hardwareintegration/client/brcm-egl/brcm-egl.pri
index eb8e450..866f179 100644
--- a/src/hardwareintegration/client/brcm-egl/brcm-egl.pri
+++ b/src/hardwareintegration/client/brcm-egl/brcm-egl.pri
@@ -11,8 +11,8 @@ for(p, QMAKE_LIBDIR_EGL) {
     exists($$p):LIBS += -L$$p
 }
 
-LIBS += $$QMAKE_LIBS_EGL
-INCLUDEPATH += $$QMAKE_INCDIR_EGL
+LIBS += -L/opt/vc/lib -lEGL -lGLESv2 -lm -lbcm_host
+INCLUDEPATH += /opt/vc/include /opt/vc/include/interface/vcos/pthreads
 
 SOURCES += $$PWD/qwaylandbrcmeglintegration.cpp \
            $$PWD/qwaylandbrcmglcontext.cpp \
diff --git a/src/hardwareintegration/compositor/brcm-egl/brcm-egl.pri b/src/hardwareintegration/compositor/brcm-egl/brcm-egl.pri
index 72645ac..8f411c5 100644
--- a/src/hardwareintegration/compositor/brcm-egl/brcm-egl.pri
+++ b/src/hardwareintegration/compositor/brcm-egl/brcm-egl.pri
@@ -16,8 +16,8 @@ for(p, QMAKE_LIBDIR_EGL) {
     exists($$p):LIBS += -L$$p
 }
 
-LIBS += $$QMAKE_LIBS_EGL
-INCLUDEPATH += $$QMAKE_INCDIR_EGL
+LIBS += -L/opt/vc/lib -lEGL -lGLESv2 -lm -lbcm_host
+INCLUDEPATH += /opt/vc/include /opt/vc/include/interface/vcos/pthreads
 
 SOURCES += \
     $$PWD/brcmeglintegration.cpp \
diff --git a/src/plugins/hardwareintegration/client/brcm-egl/brcm-egl.pro b/src/plugins/hardwareintegration/client/brcm-egl/brcm-egl.pro
index 39dcf57..6dbe537 100644
--- a/src/plugins/hardwareintegration/client/brcm-egl/brcm-egl.pro
+++ b/src/plugins/hardwareintegration/client/brcm-egl/brcm-egl.pro
@@ -5,8 +5,6 @@ QT += waylandclient-private
 
 include(../../../../hardwareintegration/client/brcm-egl/brcm-egl.pri)
 
-LIBS += -lEGL
-
 OTHER_FILES += \
     brcm-egl.json
 
diff --git a/src/plugins/hardwareintegration/compositor/brcm-egl/brcm-egl.pro b/src/plugins/hardwareintegration/compositor/brcm-egl/brcm-egl.pro
index 1608004..f47898a 100644
--- a/src/plugins/hardwareintegration/compositor/brcm-egl/brcm-egl.pro
+++ b/src/plugins/hardwareintegration/compositor/brcm-egl/brcm-egl.pro
@@ -5,8 +5,6 @@ QT = compositor compositor-private core-private gui-private
 
 OTHER_FILES += brcm-egl.json
 
-LIBS += -lEGL
-
 SOURCES += \
     main.cpp
 
diff --git a/src/plugins/platforms/qwayland-brcm-egl/qwayland-brcm-egl.pro b/src/plugins/platforms/qwayland-brcm-egl/qwayland-brcm-egl.pro
index 31bc718..1e28043 100644
--- a/src/plugins/platforms/qwayland-brcm-egl/qwayland-brcm-egl.pro
+++ b/src/plugins/platforms/qwayland-brcm-egl/qwayland-brcm-egl.pro
@@ -5,8 +5,6 @@ QT += waylandclient-private
 
 include(../../../hardwareintegration/client/brcm-egl/brcm-egl.pri)
 
-LIBS += -lEGL
-
 OTHER_FILES += \
     qwayland-brcm-egl.json
 
-- 
2.3.0

